Source/AwsCommonRuntimeKit/crt/CommonRuntimeError.swift

```diff
@@ -1,6 +1,29 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0.
 
+import AwsCCommon
+import Foundation
+
 public enum CommonRunTimeError: Error {
     case crtError(CRTError)
 }
+
+public struct CRTError: Equatable {
```

Comment:
I'm likely missing a ton of context but something doesn't seem right to essentially have two types that have the same name, except one uses the abbreviation and one does not. Is it possible to clean this up to reduce confusion and make things easier to understand?

Any reason why `CRTError` can't conform to `Error`? And perhaps we can nuke `CommonRunTimeError`.

Reply:
The idea is that we want to have CommonRuntimeError which can have both Swift as well as C errors. Currently, it doesn't have any Swift Errors but it might in the future. CRTError just encapsulates the C error. We couldn't think of a better name for CRTError. Please free to recommend if you any better suggestions.

Reply:
Would consumers need to handle C errors differently than Swift errors? Do we expect to model a Swift error differently than a C error? Could we have a single error struct that has multiple initializers to handle being initialized from C and from Swift?

Reply:
Ganesh wanted the ability to differentiate between C & Swift Errors and model them differently.

Reply:
Ok gotcha, I'll make a note of this and we can revisit cleaning up the error model when ganesh is back. I have some ideas but I also have very little context so would love ganesh's thoughts.

Reply:
I would love to clean this up too but haven't found a better way yet.